lorawan-device/src/region/dynamic_channel_plans/mod.rs

```diff
@@ -35,12 +35,25 @@ pub(crate) use in865::IN865;
 pub(crate) struct Channel {
     frequency: u32,
     _datarates: DataRateRange,
+    dl_frequency: Option<u32>,
 }
 
 impl Channel {
     /// Initialize Channel with frequency and supported minimum and maximum data rates
     fn new(f: u32, dr_min: DR, dr_max: DR) -> Self {
-        Self { frequency: f, _datarates: DataRateRange::new_range(dr_min, dr_max) }
+        Self::new_with_dr(f, DataRateRange::new_range(dr_min, dr_max))
+    }
+
+    fn new_with_dr(f: u32, dr: DataRateRange) -> Self {
```

Comment:
do we need two constructors here? Maybe it is enough to just have `new(f: u32, dr: DataRateRange)`

Reply:
I initially wanted to get rid of this as well, but the first variant is used a lot in tests.

Reply:
Id vote for fixing the tests to keep the interface simple.